util/src/main/java/io/kubernetes/client/Configmap.java

```diff
@@ -0,0 +1,197 @@
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import okhttp3.Call;
+import okhttp3.Response;
+
+/** Utility class to create Configmaps */
+public class Configmap {
+
+  private ApiClient apiClient;
+  private CoreV1Api coreClient;
+
+  public Configmap() {
+    this(Configuration.getDefaultApiClient());
+  }
+
+  public Configmap(ApiClient apiClient) {
+    this.apiClient = apiClient;
+    this.coreClient = new CoreV1Api(apiClient);
+  }
+
+  public ApiClient getApiClient() {
+    return apiClient;
+  }
+
+  public void setApiClient(ApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  /**
+   * Method to create a Configmap from a file. The file should contain key-value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param file
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromFile(String namespace, String name, Path file)
+      throws IOException, CreateConfigmapException, ApiException {
+    if (Files.isRegularFile(file)) {
+      try (BufferedReader bufferedReader =
+          new BufferedReader(
+              new InputStreamReader(Files.newInputStream(file), StandardCharsets.UTF_8.name()))) {
+        createFromReader(namespace, name, bufferedReader);
+      }
+    } else {
+      throw new CreateConfigmapException("Target path is a directory");
+    }
+  }
+
+  /**
+   * To create Configmap by passing InputStream of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param stream
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromStream(String namespace, String name, InputStream stream)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new InputStreamReader(stream));
+  }
+
+  /**
+   * To creat Configmap by passing string literal. Value should be a string in the form "key=value"
+   *
+   * @param namespace
+   * @param name
+   * @param value
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromString(String namespace, String name, String value)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new StringReader(value));
+  }
+
+  /**
+   * To create Configmap by passing Reader of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param reader
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromReader(String namespace, String name, Reader reader)
+      throws IOException, CreateConfigmapException, ApiException {
+    if (reader != null) {
+      if (name != null && !name.isEmpty()) {
+        Map<String, String> map = getPropertyMap(reader);
```

Comment:
I think we need to extend this to support any kind of file, not just `.properties` files.

Reply:
Do you mean that we identify the type of file from extension or metadata and then proceed to parse it? There is bindayData field  in V1ConfigMap. Do you think we should just pass the contents of file (as byte array) into binaryData?

Reply:
When i try to pass the binary array inside binaryData map it creates the configmap. But if you inspect the created configmap, the data is not present in it. Do you think it could be some kind of bug? There is a big chance that I might be missing something here.

Reply:
The data in the configmap is Base64 encoded. You need to Base64 encode the binary data.

The relevant Golang code is starts here:

https://github.com/kubernetes/kubectl/blob/master/pkg/generate/versioned/configmap.go#L256

Reply:
Thanks for helping me with this @brendandburns . I looked at the code of Kubectl. It helped me. I have made few changes as per your review comments. Please take a look.